plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcConfig.java

```diff
@@ -86,6 +91,34 @@ public BaseJdbcConfig setMetadataCacheTtl(Duration metadataCacheTtl)
         return this;
     }
 
+    @NotNull
+    public Duration getSchemaNamesCacheTtl()
```

Comment:
Does it make sense to cap this to be lower than metadata cache ttl? Listing operations will see a table but there would be no metadata cached for it and the table might've been dropped already.

Reply:
Lists and specific metadata objects are loaded at different times too, sot this scenario can still happen.

Reply:
@hashhar Theoretically, there could be some situations when users want to cache lists of schemas/tables for a longer amount of time (very big number of entries which do not change over time), but still let individual table data get reloaded.
This is a generic config after all.

Reply:
That's a fair point.